DigitalLearningSolutions.Data/DataServices/GroupsDataService.cs

```diff
@@ -22,6 +23,25 @@ public interface IGroupsDataService
         int? GetRelatedProgressIdForGroupDelegate(int groupId, int delegateId);
 
         void DeleteGroupDelegatesRecordForDelegate(int groupId, int delegateId);
+
+        IEnumerable<Progress> GetDelegateProgressForCourse(int delegateId, int customisationId);
+
+        void UpdateProgressSupervisorAndCompleteByDate(int progressId, int supervisorAdminId, DateTime? completeByDate);
+
+        void AddDelegateToGroup(int delegateId, int groupId, DateTime addedDate, int addedByFieldLink);
+
+        int CreateNewDelegateProgress(
+            int delegateId,
+            int customisationId,
+            int customisationVersion,
+            DateTime submittedTime,
+            int enrollmentMethodId,
+            int? enrolledByAdminId,
+            DateTime? completeByDate,
+            int supervisorAdminId
+        );
+
+        void CreateNewAspProgress(int tutorialId, int progressId);

```

Comment:
I think some of these are in the incorrect data service. Progress doesn't really relate to groups.